machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -0,0 +1,210 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=machineosbuilds,scope=Cluster,shortName=mosb
```

Comment:
Shortnames are not allowed, unless you expect the majority of cluster users to frequently access these objects. To my knowledge, this doesn't warrant that.

They create ambiguity in the CLI, since it becomes too easy to overlap. Officially, only fully qualified resource names are supported.